### ISSUE TITLE
4.x: Add metrics for concurrency limits 

### ISSUE THREAD

**Issue by tomas-langer** [helidon-io/helidon#9304]
Currently we do not have metrics in fault tolerance components, and we did not introduce them to concurrency limits (https://github.com/helidon-io/helidon/issues/8897).

We should find a good way to be able to expose information from these components that can be seen in metrics.
We should honor the name of the component, to allow differentiation between instances (i.e. we can have a fixed metric for a client and a server, and these should not be seen in metrics as the same instance) - this can be achieved through `name`, which can be specified by user in configuration, or when explicitly setting an instance.

**Comment by LateshDulani** [comment]
Watching.
Do we have a potential helidon version we expect this change to be available?

**Comment by lettrung99** [comment]
Watching.  
Thanks, 
David.

**Comment by lettrung99** [comment]
Hi team,
Could you clarify what specific types of metrics will be emitted by this enhancement? Additionally, in which version of Helidon will these new metrics features be available for users?
Thanks,
David

**Comment by vasanth-bhat** [comment]
Yes,  Having metrics is critical for uptake of concurrency limit in production. 

For Fixed Limit with queueing ,  we feel the following metrics  would be  very useful 

1.  Queue usage :  actual requests queued,    %queue full 
2.   Histogram for time spent by requests in a queue 

For AIMD :  
1. A metric that shows how the dynamic concurrency  goes up and down based on user concurrency and  requests breaching KPI

**Comment by vasanth-bhat** [comment]
Do we know which Helidon version is expected to have this fix?

**Comment by lettrung99** [comment]
Hi @m0mus - Do you have a rough ETA for Milestone 4.3.0? We are trying to plan some HPA work around the concurrency rate limits. 

Thanks in advance.
-David

**Comment by spericas** [comment]
Created specific issue for SE fault tolerance metrics. Note that MP fault tolerance metrics are already supported as part of the MicroProfile spec.
